File: ontrack-web-core/components/widgets/home/PromotionStabilityChartWidget.js
import PromotionLevelStabilityChart from "@components/promotionLevels/PromotionLevelStabilityChart";
import PromotionChartWidget from "@components/widgets/home/PromotionChartWidget";

export default function PromotionStabilityChartWidget(props) {
    return (
        <PromotionChartWidget
            prefix="Stability of"
            chart={PromotionLevelStabilityChart}
            {...props}
        />
    )
}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/dashboards/widgets/PromotionStabilityChartWidget.kt
package net.nemerosa.ontrack.model.dashboards.widgets

import org.springframework.stereotype.Component

@Component
class PromotionStabilityChartWidget: AbstractWidget<PromotionChartWidgetConfig>(
    key = "home/PromotionStabilityChart",
    name = "Promotion stability",
    description = "Chart displaying the percentage of builds being promoted.",
    defaultConfig = PromotionChartWidgetConfig(project = null, branch = null, promotionLevel = null),
    preferredHeight = 30,
)
